### ISSUE TITLE
Option to allow overriding existing import map entries in shim mode

### ISSUE THREAD

**Issue by lewisl9029** [guybedford/es-module-shims#254]
Hi again! 🙂 

In an earlier PR we added [this check](https://github.com/guybedford/es-module-shims/pull/213/files#diff-a35b52a7f96052ad4f5adb06ea0b0c5e9bd470334360d0ab8e1ce7e9b64144f4R129) to throw when users attempt to override an existing import map entry.

Fast forward to today, and I actually found a nasty edge case in my custom react-refresh workflow that requires overriding existing import map entries to fix. 😅 

Would you be open to adding an option to bypass the check? Maybe in the form of an `allowOverrides` init option that defaults to false and only takes effect in shim mode?

Happy to send over a PR if you're ok with the idea. Thanks!

**Comment by guybedford** [comment]
Hi @lewisl9029, agreed this could be a useful option to add. If you're able to work on a PR that would be great. Please just try to keep the byte size of the implementation as small as possible!

**Comment by lewisl9029** [comment]
@guybedford random idea related to your comment on byte size: wdty of potentially having a separate build for shim vs polyfill mode to further optimize byte size? It seems unlikely for a site to need to switch between two modes, and even more unlikely that it would ever work.

Might be as straightforward as replacing the shimMode export to a build-time injected boolean and reconfiguring the bundler. Will be a breaking change for users depending on automatic shim-mode detection, but I think it might be worthwhile longer term.

Happy to explore this as a separate follow up PR if that sounds like something worth exploring!

**Comment by guybedford** [comment]
@lewisl9029 if there could be a significant file saving that could be interesting, although I'm not sure it would be such a big saving as most of the code paths are shared. Separating features might well be a byte size reduction approach in future though eg having extensions loaded in based on what is needed. At the same time I quite like that it's one thing at the moment, does simplify the integration process too.

**Comment by lewisl9029** [comment]
Sounds good, I think I'll give it a shot in a quick and dirty way to explore how much we could potentially save, then we could decide if it's worthwhile once we have some actual numbers.

**Comment by lewisl9029** [comment]
PR is ready for review here: https://github.com/guybedford/es-module-shims/pull/257

Also opened a draft to test out the separate builds idea: https://github.com/guybedford/es-module-shims/pull/258

TL;DR: you were right, the savings were pretty insignificant (<0.5kb). 😅

**Comment by guybedford** [comment]
Landed in #257.
